### PR TITLE
python3Packages.typing-extensions: 4.5.0 -> 4.7.1

### PR DESCRIPTION
**Pull request by blaggacao**
###### Description of changes

https://github.com/python/typing_extensions/blob/main/CHANGELOG.md

This will trigger a mass-rebuild, hence `staging`. Help seeing this through is greatly appeciated.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/typing-extensions/default.nix [hunk 1]
```diff
@@ -32,7 +32,8 @@ buildPythonPackage rec {
   ];
 
   meta = with lib; {
-    description = "Backported and Experimental Type Hints for Python 3.5+";
+    description = "Backported and Experimental Type Hints for Python";
+    changelog = "https://github.com/python/typing_extensions/blob/main/CHANGELOG.md";
```
Comment (Kranzes) [inline pkgs/development/python-modules/typing-extensions/default.nix:36]:
```suggestion
    changelog = "https://github.com/python/typing_extensions/blob/${src.rev}/CHANGELOG.md";
```

Comment (blaggacao) [reply pkgs/development/python-modules/typing-extensions/default.nix]: Adopted in [force-push](https://github.com/NixOS/nixpkgs/compare/c34b82d178df53ac5beb6ae59a51644cef203971..538e8e2d1834ada53b51f7a31494d2b00fe8aa5a). Thx!

**Comment by trofi** [comment]
Bisect says f87ce0b9fdcc07b8f8cdc0265e600aa75d515cc4 `python3Packages.typing-extensions: 4.5.0 -> 4.7.1` broke `python310Packages.mypy` build in `staging` as:

```
$ nix build --no-link -f. python310Packages.mypy -L
...
=================================== FAILURES ===================================
___________________ StubtestUnit.test_runtime_typing_objects ___________________
[gw15] linux -- Python 3.10.12 /nix/store/cgr52a75vfdcxj0mfwrv3vr4652x9ap3-python3-3.10.12/bin/python3.10
args = (<mypy.test.teststubtest.StubtestUnit testMethod=test_runtime_typing_objects>,)
kwargs = {}
cases = [<mypy.test.teststubtest.Case object at 0x7fffeefe5330>, <mypy.test.teststubtest.Case object at 0x7fffeefe55d0>, <mypy.test.teststubtest.Case object at 0x7fffeefe5570>]
expected_errors = set()
c = <mypy.test.teststubtest.Case object at 0x7fffeefe5570>, @py_assert1 = False
@py_format3 = "{'test_module...ocol_attrs__'} == set()\n~Extra items in the left set:\n~'test_module.X.__callable_proto_members_only__'\n~'test_module.X.__protocol_attrs__'\n~Use -v to get more diff"
@py_format5 = "test_module.X.__callable_proto_members_only__\n~test_module.X.__protocol_attrs__\n~\n>assert {'test_module...ocol_att... set:\n~'test_module.X.__callable_proto_members_only__'\n~'test_module.X.__protocol_attrs__'\n~Use -v to get more diff"
output = 'test_module.X.__callable_proto_members_only__\ntest_module.X.__protocol_attrs__\n'
actual_errors = {'test_module.X.__callable_proto_members_only__', 'test_module.X.__protocol_attrs__'}

    def test(*args: Any, **kwargs: Any) -> None:
        cases = list(fn(*args, **kwargs))
        expected_errors = set()
        for c in cases:
            if c.error is None:
                continue
            expected_error = c.error
            if expected_error == "":
                expected_error = TEST_MODULE_NAME
            elif not expected_error.startswith(f"{TEST_MODULE_NAME}."):
                expected_error = f"{TEST_MODULE_NAME}.{expected_error}"
            assert expected_error not in expected_errors, (
                "collect_cases merges cases into a single stubtest invocation; we already "
                "expect an error for {}".format(expected_error)
            )
            expected_errors.add(expected_error)
        output = run_stubtest(
            stub="\n\n".join(textwrap.dedent(c.stub.lstrip("\n")) for c in cases),
            runtime="\n\n".join(textwrap.dedent(c.runtime.lstrip("\n")) for c in cases),
            options=["--generate-allowlist"],
        )

        actual_errors = set(output.splitlines())
>       assert actual_errors == expected_errors, output
E       AssertionError: test_module.X.__callable_proto_members_only__
E         test_module.X.__protocol_attrs__
E
E       assert {'test_module...ocol_attrs__'} == set()
E         Extra items in the left set:
E         'test_module.X.__callable_proto_members_only__'
E         'test_module.X.__protocol_attrs__'
E         Use -v to get more diff

mypy/test/teststubtest.py:172: AssertionError
=========================== short test summary info ============================
FAILED mypy/test/teststubtest.py::StubtestUnit::test_runtime_typing_objects
===== 1 failed, 10730 passed, 332 skipped, 10 xfailed in 312.50s (0:05:12) =====
```

**Comment by Artturin** [comment]
mypy needs to be updated  >=1.4.0 https://github.com/python/mypy/commit/b8f8d800fc1d72452af67aa0ae64c48bfacb0c92

**Pull request merged**